File: append.js
function append(arr, n) {
  if (n < 0) {
    return arr;
  }
  return [n].concat(append(arr, n-1));
}

//console.log(append([], 0));
//console.log(append([], 1));
//console.log(append([], 2));
//console.log(append([], 3));
//console.log(append([], 4));


function reverse_append(arr, n) {
  
  
  if (n < 0) {
    return arr;
  }
  
  return append(arr, n - 1).concat([n]);

}

//console.log(reverse_append([], 0));
//console.log(reverse_append([], 1));
//console.log(reverse_append([], 2));
//console.log(reverse_append([], 3));
console.log(reverse_append([], 4));

File: append.rb
require 'byebug'

def append(arr, n)
  return arr if n < 0
  debugger
  new_arr = append(arr, n -1)
  final_arr = [n] + new_arr
end

print append([],4)


def reverse_append(arr, n)
  return arr if n < 0
  
  new_arr = reverse_append(arr, n -1)
  final_arr = new_arr + [n]
end

print reverse_append([],4)